### PR TITLE
Allow each Z axis stepper to move a different direction, if needed

### LINKED ISSUE #20677: [FR] Allow non-primary Z axis to step in a different direction
### Description

Per both #13470 and #10477 (and my own problems), Marlin doesn't support secondary (etc) Z axis steppers moving the opposite direction from the primary Z motor without handling it at a wiring level.

### Feature Workflow

There should be new settings (INVERT_Z2_VS_Z_DIR, etc) to allow for this to happen.

#### Additional Information

PR coming very shortly.

### PR DESCRIPTION
**Pull request by wilbur4321**
### Requirements

This resolves #20677, #10477 and #13470 by adding configuration and code to allow secondary and greater Z steppers to step in a different direction from the primary.

### Description

This was implemented in the same was as support for secondary X and Y steppers was implemented, by having configuration settings for "should this stepper be opposite from the first Z stepper?" rather than per-motor like extruders are.  This makes the change more localized and means that no existing configurations should be broken.

### Benefits

This lets you use polarized wires without moving their pins around, making the wiring easier and look neater.

### Configurations

Attached is for four motors and `default_envs = BIGTREE_SKR_PRO`.
[configs.zip](https://github.com/MarlinFirmware/Marlin/files/5767315/configs.zip)

### Related Issues

This was built and tested for an Ender 3 Pro / BTT Skr Pro v1.1 / TMC 2208 dual-z configuration.  Building was also checked for 1, 3 and 4 stepper motor configurations, but these were of course not tested.

### CODE REVIEW

File: Marlin/Configuration_adv.h [hunk 1]
```diff
@@ -576,6 +576,14 @@
 #define NUM_Z_STEPPER_DRIVERS 1   // (1-4) Z options change based on how many
 
 #if NUM_Z_STEPPER_DRIVERS > 1
+  #define INVERT_Z2_VS_Z_DIR false   // Set 'true' if Z2 motor should rotate in opposite direction from Z
+  #if NUM_Z_STEPPER_DRIVERS > 2
+    #define INVERT_Z3_VS_Z_DIR false   // Set 'true' if Z3 motor should rotate in opposite direction from Z
+    #if NUM_Z_STEPPER_DRIVERS > 3
+      #define INVERT_Z3_VS_Z_DIR false   // Set 'true' if Z4 motor should rotate in opposite direction from Z
```
Comment (sjasonsmith) [inline Marlin/Configuration_adv.h]: This should be Z4

Comment (sjasonsmith) [reply Marlin/Configuration_adv.h]: The bottom line that is, I thought I was only commenting on one line.

Comment (wilbur4321) [reply Marlin/Configuration_adv.h]: Yikes, thank you!  Don't worry, I knew which line was wrong :) Fixed.

**Comment by thinkyhead** [comment]
But why?? You can simply flip the plug and let the code remain simple.

**Comment by sjasonsmith** [comment]
With that logic why allow inverting any motor? And please don’t remove the ability to invert motors.

It makes sense to me to treat every motor the same as far as inverting goes.

**Comment by wilbur4321** [comment]
@thinkyhead my pre-purchased motor cables are (a) colored and (b) polarized.  I actually had a cable that was modified with a reversible cable for quite a while, and just replaced it because they don't fit quite as well and work loose easier.  I could switch the pins around in the JST-PH connector well enough, but -- having the colors in the same order on every single motor makes visually checking wiring a lot easier and is rather important to lots of us with OCD/OCPD/autism.

**Comment by thinkyhead** [comment]
Thanks for your explanation. If the plug can't be flipped then it can't be flipped!

**Pull request merged**